### PR TITLE
[Workers]: Add Turso Integration

### PR DESCRIPTION
**Pull request by lauragift21**

### CODE REVIEW

File: content/workers/databases/native-integrations/turso.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
+---
+pcx_content_type: configuration
+title: Turso
+---
+
+# Turso
+
+[Turso](https://turso.tech/) is an edge-hosted, distributed database based on [libSQL](https://libsql.org/), an open-source fork of SQLite. It was designed to minimize query latency for applications where queries comes from anywhere in the world.
```
Comment (deadlypants1973) [inline content/workers/databases/native-integrations/turso.md]:
```suggestion
[Turso](https://turso.tech/) is an edge-hosted, distributed database based on [libSQL](https://libsql.org/), an open-source fork of SQLite. Turso was designed to minimize query latency for applications where queries comes from anywhere in the world.
```

File: content/workers/databases/native-integrations/turso.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+{{<render file="_database_integrations_definition.md">}}
+
+## Set up an integration with Turso
+
+To set up an integration with Turso:
+
+1. You need to install Turso CLI to create and populate a database, Use one of the following two commands in your terminal to install the Turso CLI:
```
Comment (deadlypants1973) [inline content/workers/databases/native-integrations/turso.md]:
```suggestion
1. You need to install Turso CLI to create and populate a database. Use one of the following two commands in your terminal to install the Turso CLI:
```

File: content/workers/databases/native-integrations/turso.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+  ```sh
+  # On macOS and linux with homebrew
+  $ brew install tursodatabase/tap/turso
+
+  # Manual scripted installation
+  $ curl -sSfL https://get.tur.so/install.sh | bash
+  ```
+
+  Next, Run the following command to make sure the Turso CLI is installed:
```
Comment (deadlypants1973) [inline content/workers/databases/native-integrations/turso.md]:
```suggestion
  Next, run the following command to make sure the Turso CLI is installed:
```

File: content/workers/databases/native-integrations/turso.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+  $ turso --version
+  ```
+
+2. Before you create your first Turso database, you have to authenticate with your GitHub account with the command:
+
+  ```sh
+  $ turso auth login
+
+  Waiting for authentication...
+  ✔  Success! Logged in as <your GitHub username>
```
Comment (deadlypants1973) [inline content/workers/databases/native-integrations/turso.md]:
```suggestion
  ✔  Success! Logged in as <YOUR_GITHUB_USERNAME>
```

File: content/workers/databases/native-integrations/turso.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+  $ curl -sSfL https://get.tur.so/install.sh | bash
+  ```
+
+  Next, Run the following command to make sure the Turso CLI is installed:
+
+  ```sh
+  $ turso --version
+  ```
+
+2. Before you create your first Turso database, you have to authenticate with your GitHub account with the command:
```
Comment (deadlypants1973) [inline content/workers/databases/native-integrations/turso.md]:
```suggestion
2. Before you create your first Turso database, you have to authenticate with your GitHub account by running:
```
Removing two 'with's

File: content/workers/databases/native-integrations/turso.md [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+  ```
+
+  - The libSQL client library import `@libsql/client/web` must be imported exactly as shown when working with Cloudflare workers. The non-web import will not work in the Workers environment.
+  - The `Env` interface contains the environment variable and secret you defined earlier.
+  - The `Env` interface also caches the libSQL client object and router, which are created on the first request to the Worker.
+  - The Worker uses `buildLibsqlClient` to query the `elements` database and returns the response as a JSON object.
+
+  With your environment configured and your code ready, you can now test your Worker locally before you deploy.
+
+  To learn more about Turso, refer to [Turso's official documentation](https://docs.turso.tech).
```
Comment (deadlypants1973) [inline content/workers/databases/native-integrations/turso.md]:
```suggestion
With your environment configured and your code ready, you can now test your Worker locally before you deploy.

To learn more about Turso, refer to [Turso's official documentation](https://docs.turso.tech).
```

File: content/workers/databases/native-integrations/turso.md [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+      const authToken = env.LIBSQL_DB_AUTH_TOKEN?.trim();
+      if (authToken == undefined) {
+        throw new Error("LIBSQL_DB_AUTH_TOKEN env var is not defined");
+      }
+
+      return createClient({ url, authToken })
+    }
+  ```
+
+  - The libSQL client library import `@libsql/client/web` must be imported exactly as shown when working with Cloudflare workers. The non-web import will not work in the Workers environment.
```
Comment (deadlypants1973) [inline content/workers/databases/native-integrations/turso.md]:
```suggestion
  - The libSQL client library import `@libsql/client/web` must be imported exactly as shown when working with Cloudflare Workers. The non-web import will not work in the Workers environment.
```

File: content/workers/databases/native-integrations/turso.md [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+  - The `Env` interface contains the environment variable and secret you defined earlier.
```
Comment (deadlypants1973) [inline content/workers/databases/native-integrations/turso.md]: This sentence needs a rewrite. The user did not define any environment variables, where were these? Secrets were automatically added, not defined, in step 6.

**Review by deadlypants1973 — CHANGES_REQUESTED** [review]

File: content/workers/databases/native-integrations/turso.md (hunk 8 above)
Comment (lauragift21) [reply content/workers/databases/native-integrations/turso.md]: That's right, I will rephrase the sentence.

File: content/workers/databases/native-integrations/turso.md [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+    4. Select **Integrations** > **Turso**. 
+    5. Follow the setup flow, selecting the database created in step 1.
+
+5. In your Worker, install the Turso client library:
+
+  ```sh
+  $ npm install @libsql/client
+  ```
+
+6. The following example shows how to make a query to your Turso database in a Worker. The credentials needed to connect to Turso have been automatically added as secrets to your Worker through the integration.
```
Comment (deadlypants1973) [inline content/workers/databases/native-integrations/turso.md]:
```suggestion
6. The following example shows how to make a query to your Turso database in a Worker. The credentials needed to connect to Turso have been automatically added as [secrets](/workers/configuration/secrets/) to your Worker through the integration.
```

File: content/workers/databases/native-integrations/turso.md [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+      if (authToken == undefined) {
+        throw new Error("LIBSQL_DB_AUTH_TOKEN env var is not defined");
+      }
+
+      return createClient({ url, authToken })
+    }
+  ```
+
+  - The libSQL client library import `@libsql/client/web` must be imported exactly as shown when working with Cloudflare Workers. The non-web import will not work in the Workers environment.
+  - The `Env` interface contains the environment variable and secret defined when you added the Turso integration in step 4.
```
Comment (deadlypants1973) [inline content/workers/databases/native-integrations/turso.md]:
```suggestion
  - The `Env` interface contains the [environment variable](/workers/configuration/environment-variables/) and [secret](/workers/configuration/secrets/) defined when you added the Turso integration in step 4.
```

**Review by deadlypants1973 — APPROVED** [review]

**Review by deadlypants1973 — APPROVED** [review]

**Review by tanushree-sharma — APPROVED** [review]

**Pull request merged**